### ISSUE TITLE
Header parameter content is overriden in request calls

### ISSUE THREAD

**Issue by leftys** [sonic182/aiosonic#270]
**Describe the bug**
Aiosonic request processing overrides given headers parameter. This can cause weird bugs if caller re-uses them. It might be similar with other parameters passed to requests.

**To Reproduce**
```
await client.post(post_uri, headers = self._headers, ...)
await client.get(get_uri, headers = self._headers, ...)
```

**What happens**
Headers are overridden inside the aiosonic request processing and `get` call has Content-Type header set to `application/x-www-form-urlencoded`. This causes some servers to fail, in my case of CloudFront returned mysterious 403 errors.

**Additional context**
aiosonic==0.11.3

**Comment by sonic182** [comment]
oh! a deepcopy will help here, gonna do a release whenever I get time. Thanks!

**Comment by sonic182** [comment]
@leftys it should be fixed for `aiosonic>=0.12.0` the only thing is that I didn't do a test for it, gonna reopen just to remember doing that test...
